client/src/pages/Login.js

```diff
@@ -10,11 +10,39 @@ import SexyBike from "../assets/sexyBike.svg";
 import SexyBikeRider from "../assets/sexyBikeRider.svg";
 import SexyBikeRider2 from "../assets/sexyBikeRider2.svg";
 import Micha from "../assets/micha.svg";
+import { getRiderImage, validateUser } from "../utils/api";
+import RiderSelect from "../components/helpers/RiderSelect";
+import { Link } from "react-router-dom";
 
 const Login = () => {
   const [loggedIn, setLoggedIn] = useState(false);
-  const [userName, setUserName] = useState("");
+  const [username, setUsername] = useState("");
   const [password, setPassword] = useState("");
+  const [user, setUser] = useState("");
+  const [riderImage, setRiderImage] = useState(null);
+
+  useEffect(() => {
+    const doFetch = async () => {
+      const loginstate = await validateUser({ username, password });
+      if (loginstate === true) {
+        setLoggedIn(true);
+      }
```

Comment:
setLoggedIn(!loggedIn) is a nice solution. 
But i need to check if loginstate === true, since if i just check for it to be truthy, it will always be.

Reply:
but the state is at this moment is false? 
```js
const [loggedIn, setLoggedIn] = useState(false);
```

Reply:
At the start it is false, but as soon as doFetch tries to do its work, it sets the state to something that is not false but is not true. Which leads to it not working properly as long as the state is not exactly equal to true